### PR TITLE
Add Distribution provider for ALP

### PR DESCRIPTION
**Pull request by lemon-suse**
We need add Distribution provider for ALP.

- Related ticket:  https://progress.opensuse.org/issues/123277
- Needles: N/A
- Verification run: https://openqa.opensuse.org/tests/3109492#step/podman_yast_firewall/22

### CODE REVIEW

File: lib/DistributionProvider.pm [hunk 1]
```diff
@@ -20,6 +20,7 @@ use Distribution::Sle::12;
 use Distribution::Opensuse::Leap::42;
 use Distribution::Opensuse::Leap::15;
 use Distribution::Opensuse::Tumbleweed;
+se Distribution::Alp::Alp;
```
Comment (tinawang123) [inline lib/DistributionProvider.pm:23]: Maybe miss 'u'

Comment (lemon-suse) [reply lib/DistributionProvider.pm]: Yes, copy code miss a 'u'. Thanks.

File: lib/DistributionProvider.pm [hunk 2]
```diff
@@ -20,6 +20,7 @@ use Distribution::Sle::12;
 use Distribution::Opensuse::Leap::42;
 use Distribution::Opensuse::Leap::15;
 use Distribution::Opensuse::Tumbleweed;
+use Distribution::Alp::Alp;
```
Comment (jknphy) [inline lib/DistributionProvider.pm:23]: just one Alp, we don't need a folder if there is only one file and we don't know exactly the flavors of ALP at this point.

File: lib/Distribution/Alp.pm [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+package Distribution::Alp;
+use strict;
+use warnings FATAL => 'all';
+use parent 'susedistribution';
+use YaST::Firewall::FirewallController;
+
+sub get_firewall {
```
Comment (jknphy) [inline lib/Distribution/Alp.pm:12]: we don't need this method, neither a controller.

File: products/alp/main.pm [hunk 4]
```diff
@@ -16,6 +17,7 @@ init_main();
 my $distri = testapi::get_required_var('CASEDIR') . '/lib/susedistribution.pm';
 require $distri;
 testapi::set_distribution(susedistribution->new());
+testapi::set_distribution(DistributionProvider->provide());
```
Comment (jknphy) [inline products/alp/main.pm:20]: we just need to set it once, the one 'provided' is derived class of susedistribution.

**Comment by jknphy** [comment]
in title instead of 'control' I would say is 'provider'
Please specify ticket related and if there is not needles you don't need to write it.
Verification is fine to do it with your draft for firewall but put out of the commit before merging.

File: products/alp/main.pm (hunk 4 above)
Comment (lemon-suse) [reply products/alp/main.pm:19]: updated

File: lib/Distribution/Alp.pm (hunk 3 above)
Comment (lemon-suse) [reply lib/Distribution/Alp.pm]: Updated

File: lib/Distribution/Alp.pm [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Summary: The class represents ALP distribution and provides access to
```
Comment (jknphy) [inline lib/Distribution/Alp.pm:1]: and the copyright?

Comment (lemon-suse) [reply lib/Distribution/Alp.pm:6]: Updated

**Pull request merged**